src/daemon/dbus_service.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019 Luxoft Sweden AB
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#ifndef TEMPLATE_DBUS_SERVICE_DAEMON_DBUS_SERVICE_H
+#define TEMPLATE_DBUS_SERVICE_DAEMON_DBUS_SERVICE_H
+
+#include <glibmm.h>
+
+#include "generated/dbus/template_dbus_service_common.h"
+#include "generated/dbus/template_dbus_service_stub.h"
+
+namespace TemplateDBusService::Daemon
+{
+    class DBusService final : private com::luxoft::TemplateDBusServiceStub
+    {
+    public:
+        explicit DBusService(const Glib::RefPtr<Glib::MainLoop> &main_loop);
+        ~DBusService() final;
+
+        DBusService(const DBusService &other) = delete;
+        DBusService(DBusService &&other) = delete;
+        DBusService &operator=(const DBusService &other) = delete;
+        DBusService &operator=(DBusService &&other) = delete;
+
+        void own_name();
+        void unown_name();
+
+    private:
+        void bus_acquired(const Glib::RefPtr<Gio::DBus::Connection> &connection,
+                          const Glib::ustring &name);
+        void name_acquired(const Glib::RefPtr<Gio::DBus::Connection> &connection,
+                           const Glib::ustring &name);
+        void name_lost(const Glib::RefPtr<Gio::DBus::Connection> &connection,
+                       const Glib::ustring &name);
+
+        void RemoveMeFoo(guint32 number, MethodInvocation &invocation) override;
```

Comment:
sO_wIcH_cOnVeNtIoN_aRe_We_UsIng()?

Reply:
I don't mind any convention as long as we are consistent and don't mix different styles. Perhaps this should be decided with the team, as this is a matter of personal preferences for many.

Reply:
This comes from the names in the D-Bus interface file and results in a method that must be overridden.

We can not have be consistent with the outside world since it can contradict. E.g. two libraries can require methods to be overridden and they can have different coding styles. The clang-tidy readability-identifier-naming check understands this and does not enforce the specified naming for methods that are marked with override.

Resolving this. If you want to change style we use, bring it up independently . It is like it is here for a reason.

Reply:
If you want to change this in this particular case, I guess you can send a patch to the generator to choose style (and try to add _ where appropriate to camel case names) or make gdbus-codegen-glibmm undestand something like `org.gtk.GDBus.C.Name` (see https://developer.gnome.org/gio/stable/gdbus-codegen.html ). But then you would enforce your style on users of the D-Bus XML. A bit weird.